test/unittests/t_bundle_mgr.cc

```diff
@@ -2,16 +2,75 @@
  * This file is part of the CernVM File System.
  */
 
+#include <gmock/gmock.h>
```

Comment:
Thinking about it, I'm not sure that generous mocking is the best way to go here. It is an interesting idea but in the end there is quite a bit of mechanics to mock (it's not exactly lean) and it also results in virtual methods being sprinkled around. I would prefer setting up the actual manager objects on a small, on the fly created test repository. This has been done in other unit tests, too.